core/src/main/java/io/questdb/cairo/sql/Function.java

```diff
@@ -252,6 +266,49 @@ default boolean isUndefined() {
         return getType() == ColumnType.UNDEFINED;
     }
 
+    /**
+     * DeepClone does not clone the running-states, only the init states of the function,
+     * must be called before Function::init.
+     * Used for parallel filter/groupBy execution, so that avoid Function::parser for every worker.
+     * Note: CursorFunction and WindowFunction are not supported now.
+     *
+     * @return deep clones of the function
+     */
+    default Function deepClone() {
```

Comment:
nit: might be a good idea to add a JMH test to compare the performance against full-blown SQL function compilation. This would help with further tweaks/optimizations.